### PR TITLE
Move {AnyNodeRef, AstNode} to ruff_python_ast crate root

### PR DESCRIPTION
**Pull request by konstin**
A quality of life improvement mostly mechanical refactoring.

### CODE REVIEW

**Comment by konstin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #7946** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/7946" target="_blank">Graphite</a> 
    * **PR #8011** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8011" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/8011?utm_source=stack-comment).

**Comment by konstin** [comment]
Apparently we use `AnyNode` (the non-ref version) only in tests

File: crates/ruff_python_formatter/src/type_param/type_params.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use ruff_formatter::write;
 use ruff_formatter::FormatResult;
-use ruff_python_ast::node::AstNode;
+use ruff_python_ast::AstNode;
 use ruff_python_ast::TypeParams;
```
Comment (charliermarsh) [inline crates/ruff_python_formatter/src/type_param/type_params.rs:4]: Here and elsewhere, many of these could be combined now, but I know it's hard to do automatically.

**Review by charliermarsh — APPROVED** [review]

**Review by MichaReiser — APPROVED** [review]

**Comment by charliermarsh** [comment]
Dang it, I just merged this into a non-`main` branch, I'm the worst.

**Comment by charliermarsh** [comment]
(I will fix up. I wanted to merge to avoid rebase conflicts since this changes so many files.)

**Pull request merged**